src/index.js

```diff
@@ -590,29 +592,31 @@ function createConfig(options, entry, format, writeMeta) {
 						},
 					}),
 					options.compress !== false && [
-						terser({
-							sourcemap: true,
-							compress: Object.assign(
-								{
-									keep_infinity: true,
-									pure_getters: true,
-									// Ideally we'd just get Terser to respect existing Arrow functions...
-									// unsafe_arrows: true,
-									passes: 10,
-								},
-								minifyOptions.compress || {},
-							),
-							output: {
-								// By default, Terser wraps function arguments in extra parens to trigger eager parsing.
-								// Whether this is a good idea is way too specific to guess, so we optimize for size by default:
-								wrap_func_args: false,
-							},
-							warnings: true,
-							ecma: modern ? 9 : 5,
-							toplevel: modern || format === 'cjs' || format === 'es',
-							mangle: Object.assign({}, minifyOptions.mangle || {}),
-							nameCache,
-						}),
+						options.closure && (format === 'es' || modern)
```

Comment:
We should clarify in the docs/help that Terser is always used for `cjs,iife,umd` formats, regardless of the `--closure` option.